api/v1alpha1/step_types.go

```diff
@@ -210,8 +210,8 @@ func (in Step) GetPodSpec(req GetPodSpecReq) corev1.PodSpec {
 	}
 }
 
-func (in Step) GetHeadlessServiceName(pipelineName string) string {
-	return "step-" + pipelineName + "-" + in.Spec.Name
+func (in Step) GetHeadlessServiceName() string {
+	return "step-" + in.Name
```

Comment:
surely you'll get name conflicts?

Reply:
you mean conflict with a user created service name?

Reply:
It's `in.Name`, which is already`pipelineName-stepName`, not `in.Spec.Name`.